From f4721a7545bda189732027ad22a52919dad4d715 Mon Sep 17 00:00:00 2001
From: Sujeet Baranwal <sbaranwal@marvell.com>
Date: Tue, 21 May 2019 10:48:43 -0700
Subject: [PATCH 274/386] gpio: thunderx: Adjust the msi base for cn96xx gpio

The number of cores is discovered via GPIO_CONST[PP], so formally
INTR_PIN should be at the values GPIO_CONST[PP]..GPIO_CONST[PP]+
GPIO_CONST[GPIOS]*2-1. GPIO_INT_VEC_E's INTR_PIN(0) value starts
at 0x036, leaving a gap from the last used MC_INTR_PP(23). SW
Driver compatibility expects it to start at 0x018, corresponding
to no gap. The patch addresses the inconsistency and adds
adjustment to calculate appropriate msi base for gpio for cn96xx.

Change-Id: Idb53e11ed71b76bf5c33ce7a963f992621c74cb3
Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Signed-off-by: Sujeet Kumar Baranwal <sbaranwal@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/8765
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Reviewed-on: https://sj1git1.cavium.com/9377
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/gpio/gpio-thunderx.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/gpio/gpio-thunderx.c b/drivers/gpio/gpio-thunderx.c
index f6393de6f89e..f3c7bbfd992c 100644
--- a/drivers/gpio/gpio-thunderx.c
+++ b/drivers/gpio/gpio-thunderx.c
@@ -47,6 +47,7 @@
 #define  GPIO_INTR_ENA_W1C		BIT(2)
 #define  GPIO_INTR_ENA_W1S		BIT(3)
 #define GPIO_2ND_BANK	0x1400
+#define MRVL_OCTEONTX2_96XX_PARTNUM	0xB2
 
 #define GLITCH_FILTER_400NS ((4u << GPIO_BIT_CFG_FIL_SEL_SHIFT) | \
 			     (9u << GPIO_BIT_CFG_FIL_CNT_SHIFT))
@@ -715,9 +716,9 @@ static int thunderx_gpio_probe(struct pci_dev *pdev,
 
 		ngpio = c & GPIO_CONST_GPIOS_MASK;
 
-		/* Workaround for Errata 34800 */
-		if (MIDR_IS_CPU_MODEL_RANGE(read_cpuid_id(),
-		    MIDR_MRVL_OCTEONTX2_96XX, 0x00, 0xf)) {
+		/* Workaround for all passes of T96xx */
+		if (((pdev->subsystem_device >> 8) & 0xFF)
+				== MRVL_OCTEONTX2_96XX_PARTNUM) {
 			txgpio->base_msi = 0x36;
 		} else {
 			txgpio->base_msi = (c >> 8) & 0xff;
-- 
2.17.1

